code/modules/reagents/chemistry/recipes/pyrotechnics.dm

```diff
@@ -371,3 +371,41 @@
 /datum/chemical_reaction/pyrosium/on_reaction(datum/reagents/holder, created_volume)
 	holder.chem_temp = 20 // also cools the fuck down
 	return
+
+/datum/chemical_reaction/teslium
+	name = "Teslium"
+	id = "teslium"
+	results = list("teslium" = 3)
+	required_reagents = list("stable_plasma" = 1, "silver" = 1, "blackpowder" = 1)
+	mix_message = "<span class='danger'>A jet of sparks flies from the mixture as it merges into a flickering slurry.</span>"
+	required_temp = 400
+
+/datum/chemical_reaction/reagent_explosion/teslium_lightning
+	name = "Teslium Destabilization"
+	id = "teslium_lightning"
+	required_reagents = list("teslium" = 1, "water" = 1)
+	results = list("destabilized_teslium" = 1)
+	strengthdiv = 100
+	modifier = -100
+	mix_message = "<span class='boldannounce'>The teslium starts to spark as electricity arcs away from it!</span>"
+	mix_sound = 'sound/machines/defib_zap.ogg'
+
+/datum/chemical_reaction/reagent_explosion/teslium_lightning/on_reaction(datum/reagents/holder, created_volume)
+	var/T1 = created_volume * 30						//100 units : Zap 3 times, with powers 3000/8000/15000. Tesla revolvers have a power of 10000 for comparison.
+	var/T2 = created_volume * 80
+	var/T3 = created_volume * 150
+	sleep(10)
+	if(created_volume >= 75)							//10 units minimum for lightning, 40 units for secondary blast, 75 units for tertiary blast.
+		tesla_zap(holder.my_atom, 7, T1)
+	sleep(10)
+	if(created_volume >= 40)
+		tesla_zap(holder.my_atom, 7, T2)
+	sleep(10)
```

Comment:
I don't have 30 min to waste to check whether this sleep() causes issues. I'm commenting so the OP can assure me they actually looked into this and it's not a dumb copypaste.